install_config/router/customized_haproxy_router.adoc

```diff
@@ -13,48 +13,492 @@ toc::[]
+  {{   if ne $cnList "" }}
+    acl test ssl_c_s_dn(CN) -m str {{ $cnList }}
+    http-request deny if !test
+  {{   end }}
+{{ end }}
+----
+
+You can then handle the whitelisted CNs with this command.
+
+----
+oc annotate route <route-name> --overwrite whiteListCertCommonName="CN1 CN2 CN3"
+----
+
+See xref:../../architecture/core_concepts/routes.adoc#route-specific-annotations[Route-specific Annotations]
+for more information.
+
+
+[[using-env-var]]
+=== Environment Variables
+
+The template can use whatever environment variables exist in the router pod.
+The environment variables can be set in the deployment configuration. Any
+of the existing environment variables can be used, and new ones can be added.
+
+They are referenced by the env function:
+
+----
+{{env "ROUTER_MAX_CONNECTIONS" "20000"}}
+----
+
+Where the first string is the variable, and the second string is the default
+when the variable is missing or *nil*. When `ROUTER_MAX_CONNECTIONS` is not
+set or is *nil*, 20000 is used. Environment variables are a map where the key
+is the environment variable name and the content is the value of the variable.
+
+See xref:../../architecture/core_concepts/routes.adoc#env-variables[Route-specific Environment variables]
+for more information.
+
+
+[[example]]
+
+=== Pulling It All Together, An Example
+
+Here is a simplistic template based on the HAProxy template file.
+
+Start with a comment:
+
+----
+{{/*
+  Here is a small example of how to work with templates
+  taken from the haproxy template file
+*/}}
+----
+
+The template can create any number of output files. Use a define construct
+to create an output file. The file name is specified as an argument to define,
+and everything inside the define block up to the matching end is written as
+the contents of that file.
+
+----
+{{ define "/var/lib/haproxy/conf/haproxy.config" }}
+global
+{{ end }}
+----
+
+The above will copy *global* to the */var/lib/haproxy/conf/haproxy.config* file
+and then close the file.
+
+Lets set up logging based on environment variables.
+
+----
+{{ with (env "ROUTER_SYSLOG_ADDRESS" "") }}
+  log {{.}} {{env "ROUTER_LOG_FACILITY" "local1"}} {{env "ROUTER_LOG_LEVEL" "warning"}}
+{{ end }}
+----
+
+The *env* function extracts the value for the environment variable.  If the
+environment variable is not defined or *nil*, the second argument is returned.
+
+The with construct sets the value of "." (dot) within the with block to
+whatever value is provided as an argument to with.
+The *with* action tests Dot for *nil*. If not *nil* the clause
+is processed up to the *end*. In the above assume `ROUTER_SYSLOG_ADDRESS`
+contains "/var/log/msg", `ROUTER_LOG_FACILITY` is not defined and
+`ROUTER_LOG_LEVEL` contains "info". The following will be copied to the
+output file:
+
+----
+  log /var/log/msg local1 info
+----
+
+Each admitted route ends up generating lines in the configuration file.
+Lets go through the admitted routes:
```

Comment:
"Lets go through the admitted routes" seems to informal. 
Maybe: "Each admitted route ends up generating lines in the configuration file, as follows:" or "...as shown in the following sections:"